source/parser.hera

```diff
@@ -53,11 +53,11 @@ NestedTopLevelStatements
 
 # Multiple top-level statements on the same line
 TopLevelSingleLineStatements
-  ( !EOS TopLevelStatement )+:statements ->
-    return statements.map(([, statement]) => statement)
+  TopLevelStatement+
 
 TopLevelStatement
-  _?:ws ModuleItem:statement StatementDelimiter:delimiter ->
+  # NOTE: _? allows for leading inline comments
+  !EOS _?:ws ModuleItem:statement StatementDelimiter:delimiter ->
```

Comment:
We may be able to remove `!EOS` and use `SemicolonDelimiter` since it looks like `!EOS` is only preventing the `StatementDelimiter` choices that make use of `EOS`.

Reply:
We do want to allow `EOS` (and no semicolon) at the end. I could use special code for the last statement, though this seemed a bit simpler...

Reply:
In that case maybe just add a note and remove/change the note on `TopLevelSingleLineStatements` since it's not quite clear to me.

Reply:
Revised.

If I recall correctly (this really comes from a previous PR that fixed top-level indentation), there might have also been an issue where `Expression` and hence `ModuleItem` can also start with `__`, and this prevents jumping to the next line here. Indeed, `Expression` → ... → `AssignmentExpression` → `__ AssignmentExpressionTail`. I've added this to the note too.